### PR TITLE
EDIX-20625 call slot subscribers when host resumes shells installation

### PR DESCRIPTION
**Pull request by noacoWix**
Performance measurements
---

### Method
Steps: 
1. Add the code below in `createAppHost`

initialize a flag to indicate if installing shells
CleanShot 2023-06-21 at 19 16 44@2x

accumulate execution duration of executeInstallShell just before the return statement
CleanShot 2023-06-21 at 19 17 52@2x

2. Run repluggable locally 
3. Run studio with local repluggable - Add `&debug=true` to url params + use directly or any other extension to redirect the request for repluggable to the local host
4. Refresh to load the editor
5. Open dev tools and run `window.total`

**Note** that I also disabled shell logger for these specific measurements.

### Results
Master branch
CleanShot 2023-06-21 at 19 12 34@2x

Current branch
CleanShot 2023-06-21 at 19 08 16@2x

### CODE REVIEW

**Review by itsh01 — APPROVED** [review]
Very nice 👍

File: src/API.ts [hunk 1]
```diff
@@ -343,7 +347,7 @@ export type FunctionWithSameArgs<F extends AnyFunction> = (...args: Parameters<F
  * @interface Shell
  * @extends {(Pick<AppHost, Exclude<keyof AppHost, 'getStore' | 'log' | 'options'>>)}
  */
-export interface Shell extends Pick<AppHost, Exclude<keyof AppHost, 'getStore' | 'log' | 'options'>> {
+export interface Shell extends Pick<AppHost, Exclude<keyof AppHost, 'getStore' | 'log' | 'options' | 'isInstallingEntryPoints'>> {
```
Comment (shirlynwix) [inline src/API.ts:350]: I would add getStore, log and options to the privateAppHost and exclude all of it
We can do it in a different pr

File: src/appHost.tsx [hunk 2]
```diff
@@ -511,6 +526,14 @@ miss: ${memoizedWithMissHit.miss}
         throw new Error(`API '${slotKeyToName(key)}' doesn't exist.`)
     }
 
+    function executeWhenFree(key: string, callback: () => void): void {
+        if (isInstallingEntryPoints) {
+            entryPointsInstallationEndCallbacks.set(key, callback)
```
Comment (shirlynwix) [inline src/appHost.tsx:531]: Why do we need these callbacks saved?
Won't the state update on install end be enough?

Comment (noacoWix) [reply src/appHost.tsx:531]: 1. AFAIK we do not want to relay on store update to cause re render when contributing new item.
2. Can we guarantee that `setInstalledShellNames` is called and updates the store in all possible installation flows?

**Review by shirlynwix — APPROVED** [review]

File: src/appHost.tsx (hunk 2 above)
Comment (noacoWix) [reply src/appHost.tsx:531]: After discussing this in person it was decided that it is better not to relay on the impl details of the install shells flow (meaning relay on a dispatch action that we know happens at the end of the flow), especially that we know that local state update + store update will not actually cause double re-render of all contributed item (so the cost of the additional local state update is low). 
So leaving it as is.
